### PR TITLE
Random cleanup

### PR DESCRIPTION
**Pull request by bryanloz-xilinx**
#### Problem solved by the commit
This is general code cleanup to help a future movement towards gcc12

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Missing includes
Use of C99 macros that will no longer exist
Simplify some code, that should have just been using the standard library
Use of Unique pointer to array instead of std::vector

#### How problem was solved, alternative solutions (if any) and why they were rejected
Genius was applied

#### Risks (if any) associated the changes in the commit
Limited risk

#### What has been tested and how, request additional testing if necessary
Tested by building

#### Documentation impact (if any)
None

### CODE REVIEW

**Review by stsoe — APPROVED** [review]
Thanks @bryanloz-xilinx 
We will wait for Stephen to review the xbutil changes  before merging

File: src/runtime_src/tools/xclbinutil/XclBinClass.cxx
Comment (keryell) [inline src/runtime_src/tools/xclbinutil/XclBinClass.cxx:493]: I wonder whether we really need `unsigned char` and not just `char`.
This would prebent the use of all the casts to `char*` below.

File: src/runtime_src/core/pcie/tools/xbflash.qspi/xqspips.cpp
Comment (keryell) [inline src/runtime_src/core/pcie/tools/xbflash.qspi/xqspips.cpp:22]: While you are motivated, why not sorting the whole list of includes? :-)

File: src/runtime_src/core/pcie/tools/xbflash.qspi/xspi.cpp
Comment (keryell) [inline src/runtime_src/core/pcie/tools/xbflash.qspi/xspi.cpp:26]: While you are motivated, why not sorting the whole list of includes? :-)

File: src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx
Comment (keryell) [inline src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx:340]:
```suggestion
  if (_bForceHex) {
```

**Review by keryell — COMMENTED** [review]
Diminishing our technical debt is always appreciated!

**Comment by bryanloz-xilinx** [comment]
This PR, also addresses this issue as of latest commit:
https://github.com/Xilinx/XRT/issues/6793

**Pull request merged**
